ui/views.py

```diff
@@ -52,6 +53,7 @@ def index(request):
     })
 
 
+@login_required(login_url="/")
```

Comment:
`login_url` is unnecessary here since we define [LOGIN_REDIRECT_URL](https://docs.djangoproject.com/en/1.9/ref/settings/#login-redirect-url) in `settings.py` to be the same